src/login/oidc/ClientRegistrar.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { inject, injectable } from "tsyringe";
+import { IFetcher } from "../../util/Fetcher";
+import IClient from "./IClient";
+import ILoginOptions from "../ILoginOptions";
+import IIssuerConfig from "./IIssuerConfig";
+
+export interface IClientRegistrar {
+  getClient(
+    options: ILoginOptions,
+    issuerConfig: IIssuerConfig
+  ): Promise<IClient>;
+}
+
+@injectable()
+export default class ClientRegistrar implements IClientRegistrar {
+  constructor(@inject("fetcher") private fetcher: IFetcher) {}
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private clientResponseToClientObject(response: Record<string, any>): IClient {
```

Comment:
Is there a reason this needs to be so loose?

```suggestion
  private clientResponseToClientObject(response: { clientId: string; clientSecret: string }): IClient {
```